app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt

```diff
@@ -0,0 +1,165 @@
+package org.wikipedia.notifications
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Rect
+import android.net.Uri
+import android.os.Build
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import org.wikipedia.Constants
+import org.wikipedia.R
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.notifications.NotificationPollBroadcastReceiver.Companion.getCancelNotificationPendingIntent
+import org.wikipedia.talk.TalkTopicsActivity.Companion.newIntent
+import org.wikipedia.util.DimenUtil.roundedDpToPx
+import org.wikipedia.util.ResourceUtil.bitmapFromVectorDrawable
+import org.wikipedia.util.StringUtil.fromHtml
+
+object NotificationPresenter {
+    private const val CHANNEL_ID = "MEDIAWIKI_ECHO_CHANNEL"
+
+    fun showNotification(context: Context, n: Notification, wikiSiteName: String) {
+        @DrawableRes var iconResId = R.drawable.ic_speech_bubbles
+        @ColorRes var iconColor = R.color.accent50
+
+        val builder = getDefaultBuilder(context, n.id(), n.type())
+        val title: String = fromHtml(if (n.contents != null) n.contents!!.header else "").toString()
```

Comment:
Should we use `StringUtil.fromHtml()`?

Reply:
ugh, more quirks from the automatic conversion.